cmd/libsnap-confine-private/tool.c

```diff
@@ -169,19 +199,44 @@ static int sc_open_snapd_tool(const char *tool_name) {
     return tool_fd;
 }
 
-static void sc_call_snapd_tool(int tool_fd, const char *tool_name, sc_identity identity, char **argv, char **envp) {
-    sc_call_snapd_tool_with_apparmor(tool_fd, tool_name, NULL, NULL, identity, argv, envp);
+static void sc_call_snapd_tool(int tool_fd, const char *tool_name, sc_tool_privs privs, char **argv, char **envp) {
+    sc_call_snapd_tool_with_apparmor(tool_fd, tool_name, NULL, NULL, privs, argv, envp);
 }
 
 static void sc_call_snapd_tool_with_apparmor(int tool_fd, const char *tool_name, struct sc_apparmor *apparmor,
-                                             const char *aa_profile, sc_identity identity, char **argv, char **envp) {
+                                             const char *aa_profile, sc_tool_privs privs, char **argv, char **envp) {
     debug("calling snapd tool %s", tool_name);
     pid_t child = fork();
     if (child < 0) {
         die("cannot fork to run snapd tool %s", tool_name);
     }
     if (child == 0) {
-        sc_set_effective_identity(identity);
+        /* change identity first, in case caps will drop CAP_SETUID/CAP_SETGID */
+        sc_set_effective_identity(privs.identity);
```

Comment:
consider dropping entirely, we're not giving CAP_SET{UID,GID} to helpers

Reply:
I agree, we should do this.

Reply:
This has been dropped in the next patchset